secure-api-gateway-ig-extensions/src/test/java/com/forgerock/sapi/gateway/fapi/v1/FAPIAdvancedDCRValidationFilterTest.java

```diff
@@ -0,0 +1,525 @@
+        final Response response = responsePromise.get(1, TimeUnit.SECONDS);
+        if (!response.getStatus().isSuccessful()) {
+            fail("Expected a successful response instead got: " + response.getStatus() + ", entity: " + response.getEntity().getJson());
+        }
+    }
+
+    /**
+     * Tests for the individual validators which validate particular fields within the Registration Request JWT.
+     */
+    @Nested
+    class RegistrationRequestObjectFieldValidatorTests {
+
+        private <T> void runValidationAndVerifyExceptionThrown(Validator<T> validator, T requestObject,
+                ErrorCode expectedErrorCode, String expectedErrorMessage) {
+            final ValidationException validationException = Assertions.assertThrows(ValidationException.class,
+                    () -> validator.validate(requestObject));
+            assertEquals(expectedErrorCode, validationException.getErrorCode(), "errorCode field");
+            assertEquals(expectedErrorMessage, validationException.getErrorDescription(), "errorMessage field");
+        }
+
+
+        @Test
+        void redirectUrisFieldMissing() {
+            final JsonValue missingField = json(object(field("a", "b"), field("c", "d")));
+            runValidationAndVerifyExceptionThrown(fapiValidationFilter::validateRedirectUris, missingField,
+                    ErrorCode.INVALID_REDIRECT_URI, "request object must contain redirect_uris field");
+        }
+
+        @Test
+        void redirectUrisArrayEmpty() {
+            final JsonValue emptyArray = json(object(field("a", "b"), field("c", "d"),
+                    field("redirect_uris", array())));
+            runValidationAndVerifyExceptionThrown(fapiValidationFilter::validateRedirectUris, emptyArray,
+                    ErrorCode.INVALID_REDIRECT_URI, "redirect_uris array must not be empty");
+        }
+
+        @Test
+        void redirectUrisNonHttpsUri() {
+            final JsonValue nonHttpsRedirect = json(object(field("a", "b"), field("c", "d"),
+                    field("redirect_uris", array("https://www.google.com", "http://www.google.co.uk"))));
+            runValidationAndVerifyExceptionThrown(fapiValidationFilter::validateRedirectUris, nonHttpsRedirect,
+                    ErrorCode.INVALID_REDIRECT_URI, "redirect_uris must use https scheme");
+        }
+
+        @Test
+        void redirectUrisMalformedUri() {
+            final JsonValue nonHttpsRedirect = json(object(field("a", "b"), field("c", "d"),
+                    field("redirect_uris", array("https://www.google.com", "123:@///324dfs+w34r"))));
+            runValidationAndVerifyExceptionThrown(fapiValidationFilter::validateRedirectUris, nonHttpsRedirect,
+                    ErrorCode.INVALID_REDIRECT_URI, "redirect_uri: 123:@///324dfs+w34r is not a valid URI");
+        }
+
+        @Test
+        void validRedirectUris() {
+            final JsonValue validRedirect = json(object(field("a", "b"), field("c", "d"),
```

Comment:
What's this "a": "b" stuff for?

Reply:
I'll remove, they were just some extra fields in the registration request object that the validator must ignore.